sotodlib/toast/workflows/proc_quality.py

```diff
@@ -0,0 +1,242 @@
+# Full license can be found in the top level "LICENSE" file.
+"""Timestream processing filters.
+"""
+
+import numpy as np
+from astropy import units as u
+import toast
+import toast.ops
+
+from .. import ops as so_ops
+
+
+def setup_raw_statistics(operators):
+    """Add commandline args and operators for raw timestream statistics.
+
+    Args:
+        operators (list):  The list of operators to extend.
+
+    Returns:
+        None
+
+    """
+    operators.append(toast.ops.Statistics(name="raw_statistics", enabled=False))
+
+
+def raw_statistics(job, otherargs, runargs, data):
+    """Compute timestream statistics on the raw data.
+
+    Args:
+        job (namespace):  The configured operators and templates for this job.
+        otherargs (namespace):  Other commandline arguments.
+        runargs (namespace):  Job related runtime parameters.
+        data (Data):  The data container.
+
+    Returns:
+        None
+
+    """
+    log = toast.utils.Logger.get()
+    timer = toast.timing.Timer()
+    timer.start()
+
+    # Configured operators for this job
+    job_ops = job.operators
+
+    job_ops.raw_statistics.output_dir = otherargs.out_dir
+
+    if job_ops.raw_statistics.enabled:
+        log.info_rank("Running raw statistics...", comm=data.comm.comm_world)
+        job_ops.raw_statistics.apply(data)
+        log.info_rank(
+            "Calculated raw statistics in", comm=data.comm.comm_world, timer=timer
+        )
+        job_ops.mem_count.prefix = "After raw statistics"
+        job_ops.mem_count.apply(data)
+
+
+def setup_filtered_statistics(operators):
+    """Add commandline args and operators for filtered timestream statistics.
+
+    Args:
+        operators (list):  The list of operators to extend.
+
+    Returns:
+        None
+
+    """
+    operators.append(toast.ops.Statistics(name="filtered_statistics", enabled=False))
+
+
+def filtered_statistics(job, otherargs, runargs, data):
+    """Compute timestream statistics on the filtered data.
+
+    Args:
+        job (namespace):  The configured operators and templates for this job.
+        otherargs (namespace):  Other commandline arguments.
+        runargs (namespace):  Job related runtime parameters.
+        data (Data):  The data container.
+
+    Returns:
+        None
+
+    """
+    log = toast.utils.Logger.get()
+    timer = toast.timing.Timer()
+    timer.start()
+
+    # Configured operators for this job
+    job_ops = job.operators
+
+    job_ops.filtered_statistics.output_dir = otherargs.out_dir
+
+    if job_ops.filtered_statistics.enabled:
+        log.info_rank(
+            "Running statistics on filtered data...", comm=data.comm.comm_world
+        )
+        job_ops.filtered_statistics.apply(data)
+        log.info_rank(
+            "Calculated filtered statistics in", comm=data.comm.comm_world, timer=timer
+        )
+        job_ops.mem_count.prefix = "After filtered statistics"
+        job_ops.mem_count.apply(data)
+
+
+def setup_hn_map(operators):
```

Comment:
The `hn` and `cadence` maps are more about data characterization than quality.

Reply:
Fair point, I can put those in a separate file.

Reply:
Or just rename this since noise estimation is also characterization

Reply:
Done